pennylane_lightning_gpu/src/algorithms/AdjointDiffGPU.hpp

```diff
@@ -34,6 +34,27 @@ void applyGeneratorRZ_GPU(SVType &sv, const std::vector<size_t> &wires,
     sv.applyPauliZ(wires, adj);
 }
 
+template <class T = double, class SVType>
+void applyGeneratorIsingXX_GPU(SVType &sv, const std::vector<size_t> &wires,
+                               const bool adj = false) {
+    sv.applyPauliX(std::vector<size_t>{wires.front()}, adj);
+    sv.applyPauliX(std::vector<size_t>{wires.back()}, adj);
+}
+
+template <class T = double, class SVType>
+void applyGeneratorIsingYY_GPU(SVType &sv, const std::vector<size_t> &wires,
+                               const bool adj = false) {
+    sv.applyPauliY(std::vector<size_t>{wires.front()}, adj);
+    sv.applyPauliY(std::vector<size_t>{wires.back()}, adj);
+}
+
+template <class T = double, class SVType>
+void applyGeneratorIsingZZ_GPU(SVType &sv, const std::vector<size_t> &wires,
+                               const bool adj = false) {
+    sv.applyPauliZ(std::vector<size_t>{wires.front()}, adj);
+    sv.applyPauliZ(std::vector<size_t>{wires.back()}, adj);
+}
+
```

Comment:
As discussed, I think this is fine for now.

Reply:
Should we check for `wires` size? If a vector with more than two elements, or even 1 element, is provided the first and last wires would be used and no error would come back from the call. Maybe the check is been performed in another part of the code.